examples/step-44/step-44.cc

```diff
@@ -1706,42 +1729,23 @@ namespace Step44
   {
     std::cout << "    Setting up quadrature point data..." << std::endl;
 
-    {
-      triangulation.clear_user_data();
-      {
-        std::vector<PointHistory<dim> > tmp;
-        tmp.swap(quadrature_point_history);
-      }
+    quadrature_point_history.initialize(triangulation.begin_active(),
+                                        triangulation.end(),
+                                        n_q_points);
 
-      quadrature_point_history
-      .resize(triangulation.n_active_cells() * n_q_points);
-
-      unsigned int history_index = 0;
-      for (typename Triangulation<dim>::active_cell_iterator cell =
-             triangulation.begin_active(); cell != triangulation.end();
-           ++cell)
-        {
-          cell->set_user_pointer(&quadrature_point_history[history_index]);
-          history_index += n_q_points;
-        }
-
-      Assert(history_index == quadrature_point_history.size(),
-             ExcInternalError());
-    }
-
-    // Next we setup the initial quadrature
-    // point data:
+    // Next we setup the initial quadrature point data.
+    // Note that when the quadrature point data is retrieved,
+    // it is returned as a vector of smart pointers, each of
+    // which is modifiable.
```

Comment:
"each of which is modifiable" -- not wrong, but is it worth stating? The remark seems more like a result of our discussion than something that's necessary for a user to know :-)